### PR TITLE
hydro reservoir levels : need for a new advanced parameter

### PR DESCRIPTION
**Pull request by guilpier-code**
This PR is meant to add a new hydro parameter described in ticket #125.

### CODE REVIEW

**Review by jmkerloch — CHANGES_REQUESTED** [review]
OK for me with some small changes

**Comment by guilpier-code** [comment]
GitHub detects a conflict on the merge from the current branch into develop.
Locally, the merge can be done easily : there is no conflict detected.
I suggest that we merge manually and locally, and push.

**Review by jmkerloch — APPROVED** [review]
OK for me with corrections

**Comment by guilpier-code** [comment]
After a discussion with the team, we 'd better merge **develop** into the current branch locally, and then merge this branch into develop via gitHub. By proceeding like this, the current PR is automatically closed as well as the possibly linked issues.

**Review by flomnes — COMMENTED** [review]
TL;DR : Add a switch to increase the waste coefficient to avoid wasting water, although in this case guide curves may not be enforced.

File: src/solver/hydro/daily2/h2o2_j_apply_costs.cpp [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      = problem.ProblemeHydrauliqueEtendu->ProblemeLineaireEtenduPartieFixe;
+    CorrespondanceDesVariables = problem.ProblemeHydrauliqueEtendu->CorrespondanceDesVariables;
+    NbJoursDUnProbleme = problem.ProblemeHydrauliqueEtendu->NbJoursDUnProbleme;
+
+    for (int i = 0; i < NombreDeProblemes; i++)
+    {
+        for (int j = 0; j < ProblemeLineaireEtenduPartieFixe[i]->NombreDeVariables; j++)
+            ProblemeLineaireEtenduPartieFixe[i]->CoutLineaire[j] = 0.0;
+
+        NbPdt = NbJoursDUnProbleme[i];
```
[the remark this answers is not in the document]
Comment (guilpier-code) [reply src/solver/hydro/daily2/h2o2_j_apply_costs.cpp:51]: NbPdt is already located after this for loop actually. Am I missing something ?

[the remark this answers is not in the document]
Comment (flomnes) [reply src/solver/hydro/daily2/h2o2_j_apply_costs.cpp:51]: it could be placed in the loop above 
```cpp
    for (int i = 0; i < NombreDeProblemes; i++)
    {
    NbPdt = NbJoursDUnProbleme[i];
    for (int j = 0; j < ProblemeLineaireEtenduPartieFixe[i]->NombreDeVariables; j++)
    {
    // ...
    }
    }
```

File: src/solver/hydro/daily2/h2o2_j_apply_costs.cpp [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+        for (int Pdt = 0; Pdt < NbPdt; Pdt++)
+            ProblemeLineaireEtenduPartieFixe[i]
+              ->CoutLineaire[CorrespondanceDesVariables[i]->NumeroVar_niveauxFinJours[Pdt]]
```
[the remark this answers is not in the document]
Comment (guilpier-code) [reply src/solver/hydro/daily2/h2o2_j_apply_costs.cpp:55]: CorrespondanceDesVariables[i] is already an alias for (see line 43)
